netbox/ipam/models/ip.py

```diff
@@ -866,7 +866,7 @@ def clean(self):
 
         # Check for primary IP assignment that doesn't match the assigned device/VM
         if self.pk:
-            for cls, attr in ((Device, 'device'), (VirtualMachine, 'virtual_machine')):
+            for cls, attr in ((Device, 'device'), (VirtualMachine, 'virtual_machine'), (VirtualDeviceContext, 'device')):
```

Comment:
```suggestion
            for cls, attr in ((Device, 'device'), (VirtualMachine, 'virtual_machine'), (VirtualDeviceContext, 'vdc')):
```

Reply:
The reverse attribute is vdcs and is a manytomany manager field.  I had to do some thing different but perhaps you can take a look and maybe make a suggestion to clean this up better.